CMakeLists.txt

```diff
@@ -15,7 +15,14 @@ if(CMAKE_VERSION VERSION_GREATER_EQUAL "3.20.0")
   cmake_policy(SET CMP0117 NEW)
 endif()
 
-project(opentelemetry-cpp)
+# Read the opentelemetry-cpp project and abi version from the api version header
+# file and set the third-party version tags
+include("${CMAKE_CURRENT_LIST_DIR}/cmake/version.cmake")
```

Comment:
Instead if this new version.cmake file, just hard code the defaults:

set(OPENTELEMETRY_VERSION "1.22.0")
set(OPENTELEMETRY_ABI_VERSION_DEFAULT "1")

A tbump script will maintain this, see comments on version.cmake